### PR TITLE
improve the reliability and output of rook 1.0 to 1.4 migration

### PR DESCRIPTION
**Pull request by laverya**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: scripts/common/rook.sh [hunk 1]
```diff
@@ -323,14 +291,13 @@ function rook_10_to_14() {
     # todo store these files ourselves
     curl -sSL https://raw.githubusercontent.com/rook/rook/v1.1.9/cluster/examples/kubernetes/ceph/upgrade-from-v1.0-crds.yaml | kubectl apply -f -
 
-    # todo upgrade ceph image to 14.2.5 here
     echo "Upgrading ceph to v14.2.5"
     kubectl -n rook-ceph patch CephCluster rook-ceph --type=merge -p '{"spec": {"cephVersion": {"image": "ceph/ceph:v14.2.5-20191210"}}}'
-    spinner_until -1 is_ceph_rollout_complete "14.2.5"
+    $DIR/bin/kurl rook wait-for-ceph-version "14.2.5"
```
Comment (emosbaugh) [inline scripts/common/rook.sh:296]: when something goes wrong with upgrades it is most often the case that scripts fail waiting for ceph to roll out with a timeout of 10 minutes. should we do something similar here or have the user manually kill the script? i do not have a strong opinion either way.

File: scripts/common/rook.sh [hunk 2]
```diff
@@ -399,9 +364,76 @@ EOM
     curl -sSL https://raw.githubusercontent.com/rook/rook/v1.4.9/cluster/examples/kubernetes/ceph/upgrade-from-v1.3-apply.yaml | kubectl apply -f -
     curl -sSL https://raw.githubusercontent.com/rook/rook/v1.4.9/cluster/examples/kubernetes/ceph/upgrade-from-v1.3-crds.yaml | kubectl apply -f -
     kubectl -n rook-ceph set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.4.9
-    kubectl -n rook-ceph set image deploy/rook-ceph-tools rook-ceph-tools=rook/ceph:v1.4.9
+    kubectl apply -f << EOM
```
Comment (emosbaugh) [inline scripts/common/rook.sh:367]: now that we have an add-on can we move this and other yaml to separate files?

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/common/rook.sh (hunk 1 above)
Comment (laverya) [reply scripts/common/rook.sh:296]: I can certainly see putting in a time limit, but I wasn't planning to until I had seen an issue.

File: scripts/common/rook.sh (hunk 2 above)
Comment (laverya) [reply scripts/common/rook.sh:367]: Yep, I'm working on that at the moment. (along with "actually downloading the addon and loading images from it")

Comment (laverya) [reply scripts/common/rook.sh:367]: (I already have the yaml committed, I just don't have the way to access it yet)

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**